plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/_impl.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+from typing import Any, Dict, List, Optional, Tuple, Union, cast
+
+import nevergrad as ng
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.override_parser.overrides_parser import OverridesParser
+from hydra.core.override_parser.types import (
+    ChoiceSweep,
+    IntervalSweep,
+    Override,
+    Transformer,
+)
+from hydra.core.plugins import Plugins
+from hydra.plugins.launcher import Launcher
+from hydra.plugins.sweeper import Sweeper
+from hydra.types import TaskFunction
+from omegaconf import DictConfig, ListConfig, OmegaConf
+
+from .config import OptimConf, ScalarConfigSpec
+
+# pylint: disable=logging-fstring-interpolation,no-self-used
+log = logging.getLogger(__name__)
+
+
+def verify_scalar_bounds(scalar: ng.p.Scalar) -> None:
+    if scalar.integer:
+        a, b = scalar.bounds
+        if a is not None and b is not None and b - a <= 6:
+            raise ValueError(
+                "For integers with 6 or fewer values, use a choice instead"
+            )
+
+
+def create_nevergrad_param_from_config(
+    config: Union[ListConfig, DictConfig, List[Any], Dict[str, Any], str]
+) -> Any:
+    if isinstance(config, (list, ListConfig)):
+        if isinstance(config, ListConfig):
+            config = OmegaConf.to_container(config, resolve=True)  # type: ignore
+        return ng.p.Choice(config)
+    if isinstance(config, (dict, DictConfig)):
+        specs = ScalarConfigSpec(**config)
+        init = ["init", "lower", "upper"]
+        init_params = {x: getattr(specs, x) for x in init}
+        if not specs.log:
+            scalar = ng.p.Scalar(**init_params)
+            if specs.step is not None:
+                scalar.set_mutation(sigma=specs.step)
+        else:
+            if specs.step is not None:
+                init_params["exponent"] = specs.step
+            scalar = ng.p.Log(**init_params)
+        if specs.integer:
+            scalar.set_integer_casting()
+        verify_scalar_bounds(scalar)
+        return scalar
+    return config
+
+
+def create_nevergrad_parameter_from_override(override: Override) -> Any:
+    val = override.value()
+    if not override.is_sweep_override():
+        return val
+    if override.is_choice_sweep():
+        val = cast(ChoiceSweep, val)
+        vals = [x for x in override.sweep_iterator(transformer=Transformer.encode)]
+        if "ordered" in val.tags:
+            return ng.p.TransitionChoice(vals)
+        else:
+            return ng.p.Choice(vals)
+    elif override.is_range_sweep():
+        # if shuffled or integer range and <=6, use Choice
+        vals = [x for x in override.sweep_iterator(transformer=Transformer.encode)]
+        return ng.p.Choice(vals)
+    elif override.is_interval_sweep():
+        val = cast(IntervalSweep, val)
```

Comment:
You can use:
```
assert isinstance(val, IntervalSweep)
```
instead of the cast.

Reply:
nice! I was doing the cast so mypy would stop complaining but this works and makes more sense!